floors/floors.go

```diff
@@ -135,12 +136,21 @@ func isPriceFloorsEnabledForRequest(bidRequestWrapper *openrtb_ext.RequestWrappe
 }
 
 // resolveFloors does selection of floors fields from request data and dynamic fetched data if dynamic fetch is enabled
-func resolveFloors(account config.Account, bidRequestWrapper *openrtb_ext.RequestWrapper, conversions currency.Conversions) (*openrtb_ext.PriceFloorRules, []error) {
+func resolveFloors(account config.Account, bidRequestWrapper *openrtb_ext.RequestWrapper, conversions currency.Conversions, priceFloorFetcher FloorFetcher) (*openrtb_ext.PriceFloorRules, []error) {
 	var errList []error
 	var floorRules *openrtb_ext.PriceFloorRules
 
 	reqFloor := extractFloorsFromRequest(bidRequestWrapper)
-	if reqFloor != nil {
+	if reqFloor != nil && reqFloor.Location != nil && len(reqFloor.Location.URL) > 0 {
+		account.PriceFloors.Fetcher.URL = reqFloor.Location.URL
+	}
+	account.PriceFloors.Fetcher.AccountID = account.ID
+	fetchResult, fetchStatus := priceFloorFetcher.Fetch(account.PriceFloors)
+
+	if account.PriceFloors.UseDynamicData && fetchResult != nil && fetchStatus == openrtb_ext.FetchSuccess {
```

Comment:
I agree it will avoid a extra function call, but for abstraction purpose i kept all the condition check inside the Fetch() call. This will make code more readable and flexible
Let me know your thoughts on this otherwise i will modify it

Reply:
Modified the code